frontend/javascripts/oxalis/model/bucket_data_handling/texture_bucket_manager.js

```diff
@@ -222,10 +236,12 @@ export default class TextureBucketManager {
 
   setupDataTextures(bytes: number): void {
     for (let i = 0; i < this.dataTextureCount; i++) {
+      const useFloatTexture = this.elementClass === "float";
       const dataTexture = createUpdatableTexture(
         this.textureWidth,
-        bytes * this.packingDegree,
-        THREE.UnsignedByteType,
+        // Float textures can hold a float per channel, so divide bytes by 4
+        useFloatTexture ? (bytes / 4) * this.packingDegree : bytes * this.packingDegree,
```

Comment:
Maybe you can extract this into a function `getChannelCount(byteCount, packingDegree)` or something like that? Could be defined next to `getPackingDegree` in `data_rendering_logic.js`.